src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/deletepolicy/TransportDeletePolicyAction.kt

```diff
@@ -37,6 +37,8 @@ class TransportDeletePolicyAction @Inject constructor(
         val deleteRequest = DeleteRequest(IndexManagementPlugin.INDEX_MANAGEMENT_INDEX, request.policyID)
                 .setRefreshPolicy(request.refreshPolicy)
 
-        client.delete(deleteRequest, listener)
+        client.threadPool().threadContext.stashContext().use {
```

Comment:
Can we add a comment explaining what this is doing for future reference

Reply:
copied one from alerting to our AddPolicyAction
```
/*
 * Remove security context before you call elasticsearch api's. By this time, permissions required
 * to call this api are validated.
 * Once system-indices [https://github.com/opendistro-for-elasticsearch/security/issues/666] is done, we
 * might further improve this logic. Also change try to kotlin-use for auto-closable.
 */
```